.github/workflows/codeql.yml

```diff
@@ -0,0 +1,40 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: ["main"]
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+    paths-ignore:
+      - "**.md"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      matrix:
+        language: "TODO - add the language(s) you want to analyze here, e.g. 'javascript' or 'python'"
```

Comment:
We might need to note that this uses array syntax, i.e. `['javascript']`?
Also, is it meant to be for reference or to use as is? I ask because GitHub might improve or change the workflow config, and I do not think this one will be updated should they do that.

Reply:
> I've improved it in [07d9c9f](https://github.com/mdn/project-template/commit/07d9c9fa1d6b47b3c2ed5ebcfc6473aa15f22c83), what do you think? Should we point to GH docs for the syntax?

No, I was referring to our docs on `/workflows`

Reply:
The comments you added was exactly what I meant 🎉